com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -57,49 +57,98 @@ internal struct AgentAction
+    /// component attached to the agent's [GameObject]. The <see cref="BehaviorType"/> setting
+    /// determines how decisions are made:
+    /// 
+    /// * <see cref="BehaviorType.Default"/>: decisions are made by the external process, 
+    ///   when connected. Otherwise, decisions are made using inference. If no inference model
+    ///   is specified in the BehaviorParameters component, then heuristic decision
+    ///   making is used.
+    /// * <see cref="BehaviorType.InferenceOnly"/>: decisions are always made using the trained
+    ///   model specified in the <see cref="BehaviorParameters"/> component.
+    /// * <see cref="BehaviorType.HeuristicOnly"/>: when a decision is needed, the agent's
+    ///   <see cref="Heuristic"/> function is called. Your implementation is responsible for
+    ///   providing the appropriate action.
     ///
-    /// At any step, an agent may be considered done due to a variety of reasons:
-    ///     - The agent reached an end state within its environment.
-    ///     - The agent reached the maximum # of steps (i.e. timed out).
-    ///     - The academy reached the maximum # of steps (forced agent to be done).
+    /// To trigger an agent decision automatically, you can attach a <see cref="DecisionRequester"/>
+    /// component to the Agent game object. You can also call the agent's <see cref="RequestDecision"/>
+    /// function manually. You only need to call <see cref="RequestDecision"/> when the agent is
+    /// in a position to act upon the decision. In many cases, this will be every [FixedUpdate]
+    /// callback, but could be less frequent. For example, an agent that hops around its environment
+    /// can only take an action when it touches the ground, so several frames might elapse between
+    /// one decision and the need for the next.
     ///
-    /// Here, an agent reaches an end state if it completes its task successfully
-    /// or somehow fails along the way. In the case where an agent is done before
-    /// the academy, it either resets and restarts, or just lingers until the
-    /// academy is done.
+    /// Use the <see cref="OnActionReceived"/> function to implement the actions your agent can take,
+    /// such as moving to reach a goal or interacting with its environment. Both
+    /// <see cref="CollectObservations"/> and <see cref="OnActionReceived"/> are called during the Unity
+    /// [FixedUpdate] phase.
+    /// 
+    /// When you call <see cref="EndEpisode"/> on an agent or the agent reaches its <see cref="maxStep"/> count,
+    /// its current episode ends. You can reset the agent -- or remove it from the
+    /// environment -- by implementing the <see cref="OnEpisodeBegin"/> function. An agent also
+    /// becomes done when the <see cref="Academy"/> resets the environment, which only happens when
+    /// the <see cref="Academy"/> receives a reset signal from an external process via the
+    /// <see cref="Academy.Communicator"/>.
```

Comment:
```suggestion
    /// its current episode ends. You can reset the agent by implementing the 
    /// <see cref="OnEpisodeBegin"/> function. An agent also
    /// becomes done when it is destroyed or when the <see cref="Academy"/> 
    /// resets the environment, which only happens when
    /// the <see cref="Academy"/> receives a reset signal from an external process via the
    /// <see cref="Academy.Communicator"/>.
```

You should not remove agents from the environment in the OnEpisodeBegin method. Destroying or Disabling the Agent triggers a Done automatically.

Reply:
I think your proposed wording is ambiguous; you could read it as an agent is only destroyed when receiving a reset signal.